subprojects/build-init/src/integTest/groovy/org/gradle/buildinit/plugins/BuildInitInteractiveIntegrationTest.groovy

```diff
@@ -236,4 +238,107 @@ class BuildInitInteractiveIntegrationTest extends AbstractInitIntegrationSpec {
+
+        executer.withForceInteractive(true)
+        executer.withStdinPipe()
+
+        // Initialize with no prompts
+        executer.withTasks('init', '--incubating', '--overwrite', '--dsl', 'groovy', '--type', 'basic', '--project-name', 'some-thing')
+
+        def handle = executer.start()
+        handle.waitForFinish()
+
+        then:
+        ScriptDslFixture.of(BuildInitDsl.GROOVY, targetDir, null).assertGradleFilesGenerated()
+    }
+
+    def 'prompts user to overwrite files and defaults to yes'() {
+        when:
+        // a file exists in the build dir
+        pom()
+
+        executer.withForceInteractive(true)
+        executer.withStdinPipe()
+
+        // Initialize with only overwrite prompt
+        executer.withTasks('init', '--incubating', '--dsl', 'groovy', '--type', 'basic', '--project-name', 'some-thing')
+
+        def handle = executer.start()
+        ConcurrentTestUtil.poll(60) {
+            assert handle.standardOutput.contains(String.format(overwriteFilesPrompt, executer.testDirectoryProvider.testDirectory.path))
+        }
+
+        handle.stdinPipe.write(TextUtil.platformLineSeparator.bytes)
+        handle.stdinPipe.close()
+        handle.waitForFinish()
+
+        then:
+        ScriptDslFixture.of(BuildInitDsl.GROOVY, targetDir, null).assertGradleFilesGenerated()
+    }
+
+    def 'prompts user to overwrite files and fails when answer is no'() {
+        when:
+        // a file exists in the build dir
+        pom()
+
+        executer.withForceInteractive(true)
+        executer.withStdinPipe()
+
+        // Initialize with only overwrite prompt
+        executer.withTasks('init', '--incubating', '--dsl', 'groovy', '--type', 'basic', '--project-name', 'some-thing')
+
+        def handle = executer.start()
+
+        // Select 'no'
+        ConcurrentTestUtil.poll(60) {
+            assert handle.standardOutput.contains(String.format(overwriteFilesPrompt, executer.testDirectoryProvider.testDirectory.path))
+        }
+        handle.stdinPipe.write(('no' + TextUtil.platformLineSeparator).bytes)
+
+        ConcurrentTestUtil.poll(60) {
+            assert handle.standardOutput.contains(String.format(overwriteFilesPrompt, executer.testDirectoryProvider.testDirectory.path))
+        }
+
+        handle.stdinPipe.write(TextUtil.platformLineSeparator.bytes)
+        handle.stdinPipe.close()
+        def result = handle.waitForFailure()
+
+        then:
+        result.assertHasDescription('Execution failed for task \':init\'')
+        result.assertHasCause(String.format(initOverwriteFilesExceptionMessage, executer.testDirectoryProvider.testDirectory.path))
+
+        expect:
+        ScriptDslFixture.of(BuildInitDsl.GROOVY, targetDir, null).assertGradleFilesNotGenerated()
+    }
+
+    def 'prompts user to overwrite files and continues when answer is yes'() {
+        // a file exists in the build dir
+        pom()
+
+        executer.withForceInteractive(true)
+        executer.withStdinPipe()
```

Comment:
These lines setting up the pom and and stdin and repeated in all the tests, as are the teardown tests closing and cleaning up the pipe.  Perhaps they should be extracted to private methods, or make use of Spocks's `setup` and `teardown`?